### ISSUE TITLE
Allow new plugins to extend @App.view decorator parameters

### ISSUE THREAD

**Issue by kagesenshi** [morepath/morepath#530]
Currently I did not see an obvious way to extend @App.view 

This would be useful for developing plugins that deal with querying available views such as:

- CORS configuration per view
- Swagger / OpenAPI JSON generator 

overriding App.view with a classmethod might be one route a person can take to preprocess kwargs before view registration, but i think it wont work in the case of multiple plugins want to use @App.view kwargs ..

**Comment by kagesenshi** [comment]
Eg: 

currently one can register more.cors per-view override by running

```python
App.cors(model=Root, view_name='myview',
         allowed_headers=['Cache-Control'], expose_headers=['Cookie'],
         allowed_origin='http://localhost.localdomain', allow_credentials=False,
         max_age=10)

@App.json(model=Root, view_name='myview')
def myview(context, request):
      return {'message': 'Hello World'}
```

but it would be nice if we can extend App.view to allow this instead

```python

@App.view(model=Root, view_name='myview',
         allowed_headers=['Cache-Control'], expose_headers=['Cookie'],
         allowed_origin='http://localhost.localdomain', allow_credentials=False,
         max_age=10)
def myview(context, request):
      return {'message': 'Hello World'}
```

**Comment by href** [comment]
I agree that this would definitely be useful. I for one would like to add a CSRF check to a number of my views, ideally triggered by a simple custom keyword argument.

I was reminded of your proposal today when I read about [Pyramid's view derivers](https://docs.pylonsproject.org/projects/pyramid/en/latest/narr/hooks.html#view-derivers). It basically allows for custom keywords to be added to views. In Pyramid's specific way of course, but like your proposal in spirit.

We could add this for views. So maybe we could have a way to register keywords for specific directives. This might also be useful for other directives if we could make it into a general concept. Those keyword-introducing directives could then be overwritten/extended again by sub-applications.
